3529/1/data_finder.rb

```diff
@@ -0,0 +1,85 @@
+class DataFinder
+  attr_reader :data_about_gem
+
+  def initialize(gem_name, url, client)
+    @data_about_gem = {}
+    @data_about_gem[:name] = gem_name
+    @url = url
+    @client = client
+    @repo_addr = adress_handler
```

Comment:
Don't assign inside method to instance variable inside initializer. You should have here only instance variables that assigns from parameters and rarely other classes calls

Reply:
If that method use data from another class and you use it a few time, then use `memoization`